lib/rails/action_mailer/tracer.rb

```diff
@@ -0,0 +1,51 @@
+module ActionMailer
+  module Tracer
+    class << self
+      def instrument(tracer: OpenTracing.global_tracer, active_span: nil)
+        clear_subscribers
+        @subscriber = ::ActiveSupport::Notifications.subscribe('deliver.action_mailer') do |*args|
+            ActionMailer::Tracer.mail(tracer: tracer, active_span: active_span, args: args)
+        end
+      end
+
+      def disable
+        if @subscriber
+          ActiveSupport::Notifications.unsubscribe(@subscriber)
+          @subscriber = nil
+        end
+      end
+      alias :clear_subscribers :disable
+
+      def mail(tracer: OpenTracing.global_tracer, active_span: nil, args:)
+        _, start, finish, _, payload = *args
+
+        span = start_span(payload.fetch(:mailer),
+                          tracer: tracer,
+                          active_span: active_span,
+                          start_time: start,
+                          message_id: payload.fetch(:message_id),
```

Comment:
what about other information like data, bcc etc which is supplied within the event? I assume that was the only information necessary for your use case?